parts/k8s/kubernetesmastercustomscript.sh

```diff
@@ -89,6 +93,45 @@ else
     echo "CA_PRIVATE_KEY is empty, assuming worker node"
 fi
 
+ETCD_SERVER_PRIVATE_KEY_PATH="/etc/kubernetes/certs/etcdserver.key"
+touch "${ETCD_SERVER_PRIVATE_KEY_PATH}"
+chmod 0600 "${ETCD_SERVER_PRIVATE_KEY_PATH}"
+chown root:root "${ETCD_SERVER_PRIVATE_KEY_PATH}"
+echo "${ETCD_SERVER_PRIVATE_KEY}" | base64 --decode > "${ETCD_SERVER_PRIVATE_KEY_PATH}"
+
+ETCD_CLIENT_PRIVATE_KEY_PATH="/etc/kubernetes/certs/etcdclient.key"
+touch "${ETCD_CLIENT_PRIVATE_KEY_PATH}"
+chmod 0644 "${ETCD_CLIENT_PRIVATE_KEY_PATH}"
```

Comment:
Is this safe? Either we leave it as 0600 and then etcdctl doesn't work since the key needs permission (so sudo etcdctl is required instead) or use 0644 to be able to read the key for etcdctl commands. @jackfrancis

Reply:
What user context does etcdctl run under? If we assign file ownership of the private key to that user, then `0600` should work.

Reply:
True, right now I'm assigning it to etcd in custom data but I think it should work if I leave it as root. I'll test + push the change.